adapter/noop2/noop2.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 Istio Authors.
```

Comment:
Creating the adapter and configuring in a separate step is not great. From a consistency/reliability perspective, it is preferable to lean towards an immutable model. An adapter object is created and initialized in one step and never changes afterwards. This prevents there being the potential for partially initialized objects floating around.

So can we just have "Validate" to validate a config block, and "Create" that takes a config block and returns a fully-baked adapter object?